src/hooks/useUrlPagination/index.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { useNavigate } from 'hooks/useNavigate';
+
+export type UseUrlPaginationOutput = {
+  currentPage: number;
+  setCurrentPage: (newPageIndex: number) => void;
+};
+
+export const PAGE_PARAM_NAME = 'page';
+
+export const useUrlPagination = (): UseUrlPaginationOutput => {
+  const { navigate } = useNavigate();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const pageIndex = searchParams.get(PAGE_PARAM_NAME) ?? undefined;
+
+  const setPageIndex = useCallback(
+    (newPageIndex: string | number) =>
+      setSearchParams(
+        currentSearchParams => ({
+          ...Object.fromEntries(currentSearchParams),
+          [PAGE_PARAM_NAME]: String(newPageIndex),
+        }),
+        {
+          replace: true,
+        },
+      ),
+    [setSearchParams],
+  );
+
+  useEffect(() => {
+    // Add page param to URL if none has been set
+    if (!pageIndex) {
+      // Note: although the pagination starts from 0, we make it start from 1 in the
+      // URL to make it more user-friendly. This is something we need to account for
+      // when updating the page search param
+      setPageIndex(1);
+    }
+
+    // Scroll to the top of the page on search change
+    window.scrollTo(0, 0);
```

Comment:
Search param changes aren't caught by the root scroll to the top [here](https://github.com/VenusProtocol/venus-protocol-interface/pull/1941/files#diff-b226a971b851720f6b199049faf0b035d75c99ff449f9981070491ce7033a76fR43)

Reply:
Good point. We can actually remove this logic, as React Router v6 scrolls to the top of the page on param change by default